src/metabase/upload.clj

```diff
@@ -170,18 +173,57 @@
     "unnamed_column"
     (u/slugify (str/trim raw-name))))
 
+(defn- is-pk?
+  [[col-name type]]
+  (and (#{"id" "pk"} (u/lower-case-en (name col-name)))
+       (isa? type ::int)))
```

Comment:
The only constraint for a PK column is that the column has an integer type. So we can't assume that the column in the table we create should be auto-incrementing. And yet we do [here](https://github.com/metabase/metabase/pull/34173/files#r1348384783)